docs/probe.md

```diff
@@ -0,0 +1,177 @@
+    - `my_model` matches input in `probe = get_probe("my_model")`
+    - `layer_output` matches variable name in `probe.update(layer_output=x_out)`
+
+#### Step 4
+For the scope of this document, we don't dwell on what `armory` is doing with `get_hub().connect_meter(meter)` other than to mention this step is necessary for establishing the connection between `meter` created in `armory/user_init.py` and `probe` created in the modified version of `armory/baseline_models/pytorch/cifar.py`.
+
+### Config Setup
+Last but not least, the config file passed to `armory run` needs to updated for these changes to take effect. Assuming the `"model"` block has been changed appropriately, the other block that needs to be added is `"user_init"`:
+```json
+...
+    "model": {
+        ...
+    },
+    "user_init": {
+        "module": "user_init",
+        "name": "user_init_function"
+    },
+...
+```
+This will prompt armory to run `user_init_function` in `user_init.py` before anything else is loaded for the scenario.
+
+## Example 2: Attack Output
+### User Story
+I defined a custom attack with `CARLADapricotPatch` in `armory/custom_attack.py`, and I am interested in the patch after <ins>***every iteration***</ins>, which is generated by `CARLADapricotPatch._augment_images_with_patch` and returned as an output.
+### `Probe` Example Code
+```python
+from armory.art_experimental.attacks.carla_obj_det_patch import CARLADapricotPatch
+from armory.instrument import get_probe
+probe = get_probe("my_attack")
+
+class CustomAttack(CARLADapricotPatch):
+    def _augment_images_with_patch(self, **kwargs):
+        return_value = super()._augment_images_with_patch(**kwargs)
+        x_patch, patch_target, transformations = return_value
+        probe.update(attack_output=x_patch)
+
+        return return_value
+```
+#### Step 1
+This step is the same as before, except `Probe` name is set to`"my_attack"`, which is what the user can refer to later to apply processing functions through a `Meter` object.
+
+#### Step 2
+The only difference between `CustomAttack` and `CARLADapricotPatch` is that `_augment_images_with_patch` has been redefined to call on `CARLADapricotPatch._augment_images_with_patch` and then have `probe` update the value for `x_patch` that results from that call. An updated value of `x_patch` is stored in `attack_output` via `probe.update(attack_output=x_patch)`. Like the `Probe` name `"my_attack"`, `attack_output` can be referenced by the user later to apply additional processing functions through a `Meter` object.
+
+### `Meter` Example Code
+Now that a `Probe` instance has been created, we need to create a `Meter` object to accept any updated values from `Probe` and apply further processing that the user desires. Suppose the user created a script located at `armory/user_init.py`:
+```python
+from armory.instrument import get_hub, Meter
+
+def user_init_function():
```

Comment:
rename `user_init_function()` to something more descriptive a la `create_meter()` or `set_up_meter()`?

Reply:
and same for above

Reply:
resolved with 89d200a6